src/main/scala/com/gwi/api/TaskDetail.scala

```diff
@@ -0,0 +1,44 @@
+package com.gwi.api
+
+import com.gwi.Main.ServerUri
+import com.gwi.execution.Task
+import io.circe.Encoder
+import io.circe.generic.semiauto.deriveEncoder
+
+import java.time.Instant
+import java.util.UUID
+
+object TaskDetail {
+  implicit val taskDetailResponseEncoder: Encoder[TaskDetail] = deriveEncoder
```

Comment:
I personally believe such encoders/decoders are rather part of infra/protocols and should be put over there. :)

Reply:
I agree, the way I chosen needs maybe a bit less boilerplate in a first place and in the previous project I worked on, it was done this way but I believe to have it separated is cleaner.
In the last commit, I extracted codecs to a separate file, merged model classes to one package object and also separated transformations to the standalone file. I think it's better now, thanks for pointing that out.